### PR TITLE
Fixed the issue #2535 Added user followers and following in class User and added test cases for it

### PR DESCRIPTION
**Pull request by Amrit02102004**
image
image

Issue : #2535 

Added the variable in docs also for future reference

Fix https://github.com/huggingface/huggingface_hub/issues/2535.

### CODE REVIEW

File: tests/test_hf_api.py [hunk 1]
```diff
@@ -4074,6 +4074,10 @@ def test_user_overview(self) -> None:
         assert overview.num_upvotes > 10
         assert len(overview.orgs) > 0
         assert any(org.name == "huggingface" for org in overview.orgs)
+        assert isinstance(overview.num_following, (int, type(None)))
+        assert overview.num_following is None or overview.num_following >= 0
+        assert isinstance(overview.num_followers, (int, type(None)))
+        assert overview.num_followers is None or overview.num_followers >= 0
```
Comment (Wauplin) [inline tests/test_hf_api.py:4080]:
```suggestion
        assert overview.num_following > 300
        assert overview.num_followers > 1000
```

We want to have a meaningful test. In our case, `julien-c` has many following and followers so it's preferable to test that. I realized the API is not yet released (should be the case in 1-2 days), that's why you are currently getting `None`. Let's wait a bit and we'll merge once the API is proded.

**Review by Wauplin — APPROVED** [review]
Thanks for the quick PR @Amrit02102004! Looks good to me except that you've been faster than the deployment process :smile: Let's wait 1-2 days before merging this just to be sure.

**Comment by Amrit02102004** [comment]
Thanks for the quick merge 😊

**Comment by Wauplin** [comment]
The feature has been proded so let's merge this. Thanks again @Amrit02102004! :hugs:

**Pull request merged**